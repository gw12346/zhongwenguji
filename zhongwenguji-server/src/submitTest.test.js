const express = require('express');
const supertest = require('supertest');
const submitTest = require('./submitTest');

describe('submitTest', () => {
  it('calculates curve parameters', () => {
    const app = express();
    app.post('/', submitTest());
    return supertest(app)
      .post('/')
      .send({
        testData: [
          { isTested: true, knownPercent: 100, range: [0, 250] },
          { isTested: false, knownPercent: NaN, range: [250, 500] },
          { isTested: true, knownPercent: 50, range: [500, 750] },
          { isTested: true, knownPercent: 10, range: [750, 1000] },
          { isTested: true, knownPercent: 0, range: [1000, 1250] },
          { isTested: true, knownPercent: 0, range: [1250, 1500] },
          { isTested: false, knownPercent: NaN, range: [1500, 1750] },
          { isTested: false, knownPercent: NaN, range: [1750, 2000] }
        ]
      })
      .expect(200)
      .then(({ body }) => {
        expect(typeof body.curveParams.amplitude).toBe('number');
        expect(typeof body.curveParams.decayStartX).toBe('number');
        expect(typeof body.curveParams.decayPeriod).toBe('number');
        expect(body.curveParams).toMatchSnapshot();
      });
  });
});
