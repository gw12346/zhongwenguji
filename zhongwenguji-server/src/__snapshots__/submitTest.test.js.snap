// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`submitTest calculates curve parameters 1`] = `
Object {
  "amplitude": 99.99999998743786,
  "decayPeriod": 207.5871400793618,
  "decayStartX": 521.2064300719244,
}
`;
